include/alpaka/kernel/Traits.hpp

```diff
@@ -224,9 +251,7 @@ namespace alpaka
 #else
         static_assert(std::is_trivially_copyable_v<TKernelFnObj>, "Kernels must be trivially copyable!");
 #endif
-        static_assert(
-            (std::is_trivially_copyable_v<std::decay_t<TArgs>> && ...),
-            "Kernel arguments must be trivially copyable!");
+        detail::assertArgsTriviallyCopyable<std::decay_t<TArgs>...>();
```

Comment:
The compiler says that the lambda is not trivially copyable, which is surprising, because the lambda in the reduce example:
```c++
auto addFn = [] ALPAKA_FN_ACC(T a, T b) -> T { return a + b; };
```
certainly looks trivially copyable to me.

Reply:
haha, the C++ standard leaves it implementation defined whether a lambda is trivially copyable or not: https://eel.is/c++draft/expr.prim.lambda#closure-2.2. That does not explain why it worked before and now doesn't. But technically, we should exempt lambdas in the check here.

Reply:
I found the same, that it is implementation behavior if a lambda is trivially copyable. I change the code to a generator expression and used my helper class, all work as soon as I use the fold expression.

```C++
   namespace detail
   {
        template<typename THead>
        struct AssertTriviallyCopyable: std::is_trivially_copyable<THead>
        {
            static_assert(std::is_trivially_copyable_v<THead>, "Kernels N-th argument must be trivially copyable!");
        };



    } // namespace detail
        static_assert(
        (detail::AssertTriviallyCopyable<std::decay_t<TArgs>>::value && ...),
            "Kernel arguments must be trivially copyable!");
```